subprojects/gradle/build-metrics/src/main/kotlin/com/avito/android/plugin/build_metrics/internal/cache/BuildCacheOperationListener.kt

```diff
@@ -4,83 +4,133 @@ import com.avito.android.gradle.metric.BuildEventsListener
 import com.avito.android.gradle.metric.NoOpBuildEventsListener
 import com.avito.android.plugin.build_metrics.internal.AbstractBuildOperationListener
 import com.avito.android.plugin.build_metrics.internal.buildOperationListenerManager
-import com.avito.android.stats.CountMetric
-import com.avito.android.stats.SeriesName
-import com.avito.android.stats.StatsDSender
+import com.avito.android.plugin.build_metrics.internal.dump
+import com.avito.android.sentry.environmentInfo
 import com.avito.android.stats.statsd
 import org.gradle.api.Project
+import org.gradle.api.Task
 import org.gradle.api.internal.project.ProjectInternal
-import org.gradle.api.provider.Provider
-import org.gradle.caching.http.HttpBuildCache
+import org.gradle.api.internal.tasks.TaskExecutionOutcome
+import org.gradle.api.internal.tasks.execution.ExecuteTaskBuildOperationDetails
+import org.gradle.api.internal.tasks.execution.ExecuteTaskBuildOperationType
 import org.gradle.caching.internal.controller.operations.LoadOperationDetails
 import org.gradle.caching.internal.controller.operations.StoreOperationDetails
+import org.gradle.caching.internal.operations.BuildCacheRemoteLoadBuildOperationType
+import org.gradle.execution.RunRootBuildWorkBuildOperationType
+import org.gradle.internal.operations.BuildOperationCategory
 import org.gradle.internal.operations.BuildOperationDescriptor
 import org.gradle.internal.operations.OperationFinishEvent
+import org.gradle.internal.operations.OperationIdentifier
+import java.util.concurrent.ConcurrentHashMap
 
 internal class BuildCacheOperationListener(
```

Comment:
big amount of private methods signals about delegation refactorting to me

Reply:
Agree, thought about it. Was not sure about side effects and shared state.
Maybe after adding next metrics use cases will more clear.